app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+				$args = $this->decorate_taxonomy_args($field, $args, $request );
+			}
+			if ($field[self::FIELD_INDEX_TYPE] === Date_Field::NAME) {
+				$args = $this->decorate_date_args($field, $args, $request );
+			}
+
+			if ($field[self::FIELD_INDEX_TYPE] === Select_Field::NAME) {
+				$args = $this->decorate_select_args($field, $args, $request );
+			}
+		}
+
+		// error_log(print_r( $args, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		// error_log(print_r( $request, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+		// error_log(print_r( $fields, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+		// error_log(print_r( $args, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		return $args;
+
+	}
+
+	public function custom_register_rest_fields() {
+
+		register_rest_field( Archive_Item::NAME, 'feature_image', array(
+			'get_callback' => function( $arr ) {
+				$post_thumbnail_id = get_post_thumbnail_id( $arr['id'] );
+				// return $post_thumbnail_id;
+				return $this->get_image_data_for_api($post_thumbnail_id);
+			}
+		) );
+
+		register_rest_field( Archive_Item::NAME, 'permalink', array(
+			'get_callback' => function( $arr ) {
+				return get_the_permalink($arr['id']);
+			}
+		) );
+
+		// only do this extra if the popup is used.
+		$has_popup = carbon_get_theme_option(Settings::FIELD_GENERAL_BROWSE_MODAL);
+		if (!$has_popup) {
+			return;
+		}
+		// error_log(print_r( $this->fields, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		// Taxonomies are already added
+
+		// add CPT field vars
+		$fields = $this->get_fields();
+		$cpt_fields = array_filter($fields, function($field) {
+			return $field[self::FIELD_INDEX_TYPE] === CPT_Field::NAME;
+		});
+		if ($cpt_fields) {
+			register_rest_field( Archive_Item::NAME, 'cpts', array(
+				'get_callback' => function( $arr ) use( &$cpt_fields) {
+					$ret = [];
+					// error_log(print_r( $cpt_fields, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+					foreach($cpt_fields as $field) {
+						$field_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_ID );
+						// $ret[$field_cpt_id] = carbon_get_post_meta( $arr['id'],FieldPostMeta::FIELD_ID );
+
+						// $type = carbon_get_post_meta( $cptid, Post_Meta::FIELD_TYPE );
+						$ret[$field_id] = carbon_get_post_meta( $arr['id'], $field_id);
+
+					}
+					return $ret;
+				}
+			) );
+		}
+
+		// add Selects field vars
+		$select_fields = array_filter($fields, function($field) {
+			return $field[self::FIELD_INDEX_TYPE] === Select_Field::NAME;
+		});
+		if ($select_fields && count($select_fields)) {
+			register_rest_field( Archive_Item::NAME, 'selects', array(
+				'get_callback' => function( $arr ) use( &$select_fields) {
+					$ret = [];
+					foreach($select_fields as $field) {
+						$field_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_ID );
+						$ret[$field_id] = carbon_get_post_meta( $arr['id'], $field_id);
```

Comment:
check if empty here and elsewhere this logic is applied.

Reply:
I updated this code